File: document/src/vespa/document/update/clearvalueupdate.cpp
// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.

#include "clearvalueupdate.h"
#include <vespa/vespalib/util/xmlstream.h>
#include <ostream>

using namespace vespalib::xml;

namespace document {

IMPLEMENT_IDENTIFIABLE(ClearValueUpdate, ValueUpdate);

bool
ClearValueUpdate::operator==(const ValueUpdate& other) const
{
    return (other.getClass().id() == ClearValueUpdate::classId);
}

// Ensure that this update is compatible with given field.
void
ClearValueUpdate::checkCompatibility(const Field&) const
{
}

// Apply this update to the given document.
bool
ClearValueUpdate::applyTo(FieldValue& ) const
{
    return false;
}

void
ClearValueUpdate::printXml(XmlOutputStream& xos) const
{
    xos << XmlTag("clear") << XmlEndTag();
}

// Print this update in human readable form.
void
ClearValueUpdate::print(std::ostream& out, bool, const std::string&) const
{
    out << "ClearValueUpdate()";
}

// Deserialize this update from the given buffer.
void
ClearValueUpdate::deserialize(const DocumentTypeRepo&, const DataType&, nbostream &)
{
}

}

File: document/src/main/java/com/yahoo/document/update/ClearValueUpdate.java
// Copyright Vespa.ai. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
package com.yahoo.document.update;

import com.yahoo.document.DataType;
import com.yahoo.document.datatypes.FieldValue;
import com.yahoo.document.serialization.DocumentUpdateWriter;

/**
 * <p>Value update that represents clearing a field. Clearing a field mean removing it.</p>
 * @author <a href="mailto:einarmr@yahoo-inc.com">Einar M R Rosenvinge</a>
 */
public class ClearValueUpdate extends ValueUpdate {

    public ClearValueUpdate() {
        super(ValueUpdateClassID.CLEAR);
    }

    @Override
    public FieldValue applyTo(FieldValue fval) {
        return null;
    }

    @Override
    protected void checkCompatibility(DataType fieldType) {
        // empty
    }

    @Override
    public FieldValue getValue() {
        return null;
    }

    @Override
    public void setValue(FieldValue value) {
        // empty
    }

    @Override
    public void serialize(DocumentUpdateWriter data, DataType superType) {
        data.write(this, superType);
    }
}
